eCommerce/security/signals.py

```
@@ -0,0 +1,28 @@
+from django.db.models.signals import post_save, post_delete
+from django.dispatch import receiver
+from django.core.cache import cache
+from .models import BlacklistedIP
+
+# The key for the Redis set that stores blacklisted IPs
+BLACKLIST_CACHE_KEY = "blacklisted_ips"
+
+
+@receiver(post_save, sender=BlacklistedIP)
+def add_ip_to_blacklist(sender, instance, **kwargs):
```

Ruff (0.14.7)

Unused function argument: `sender`

(ARG001)